controller/src/main/java/org/jboss/as/controller/ParsedBootOp.java

```diff
@@ -84,6 +85,21 @@ public PathAddress getAddress() {
         return address;
     }
 
+    public boolean isBootHandlerUpdateNeeded() {
+        return this.handler instanceof ParallelBootOperationStepHandler
+                && (
+                    bootHandlerUpdateNeeded
+                    || ((ParallelBootOperationStepHandler) this.handler).getParsedBootOp().getChildOperations().size() != getChildOperations().size());
```

Comment:
Is this last line still relevant?

I'm quite lost on the condition that would like to this evaluating true, but I think it has something to do with YamlConfigurationExtension.RemoveOperation.processOperation.

But would that ever get called in a situation where 1) handler instanceof ParallelBootOperationStepHandler and 2) YamlConfigurationExtension.processOperations hadn't called bootHandlerUpdateNeeded()? AIUI now bootHandlerUpdateNeeded() always gets called if there's a configExtension and parallel boot is happening.

Leaving this line is "safer" in case I misunderstood something and you wrongly believe me. ;) But otherwise all this is really hard to understand so if this bit of complexity can be removed it helps a bit.

Reply:
correct